java/src/main/java/io/opentracing/contrib/ServerTracingInterceptor.java

```diff
@@ -83,8 +82,15 @@ public ServerServiceDefinition intercept(BindableService bindableService) {
     ) {
         Map<String, String> headerMap = new HashMap<String, String>();
         for (String key : headers.keys()) {
-            String value = headers.get(Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER));
-            headerMap.put(key, value);
+            if (key.endsWith(Metadata.BINARY_HEADER_SUFFIX)) {
```

Comment:
I'd argue that we should just drop these... WDYT? Especially since `byteValue` may not even be valid `UTF-8`, thus leading to undefined behavior.

I greatly dislike how the OT user now needs to "know" about this weird `-bin` suffix semantic, btw... but not sure how to document that. It's some comfort that, in practice, it's unlikely to crop up. But only because of its obscurity.